### PR TITLE
Add Roslyn analyzer package support

### PR DESCRIPTION
**Pull request by nowsprinting**
Roslyn analyzers supported on Unity 2020.2 or later.

The analyzer needs to make the following settings in plugin import settings.
- Off all platforms
- Set label `RoslynAnalyzer`

With this setting, it will be enabled as an analyzer in Unity 2020.2 or later, and enabled in generated .csproj by the following plugins.
- JetBrains Rider Editor v2.0.6+
- Visual Studio Code Editor v1.2.0+

see more information
https://docs.unity3d.com/2020.2/Documentation/Manual/roslyn-analyzers.html

### CODE REVIEW

File: Assets/NuGet/Editor/NugetHelper.cs [hunk 1]
```diff
@@ -1448,12 +1448,37 @@ public static bool Install(NugetPackage package, bool refreshAssets = true)
...
+                    ModifyImportSettingsIfRoslynAnalyzer(package.Id, package.Version);
                     EditorUtility.ClearProgressBar();
                 }
             }
             return installSuccess;
         }
 
+        private static void ModifyImportSettingsIfRoslynAnalyzer(string id, string version)
+        {
+            var dir = $"Assets/Packages/{id}.{version}";
```
Comment (robin-moss) [inline Assets/NuGet/Editor/NugetHelper.cs:1460]: You should use `NugetConfigFile.RepositoryPath` instead of the hardcoded path

Comment (robin-moss) [inline Assets/NuGet/Editor/NugetHelper.cs:1460]:
```suggestion
            var dir = Path.Combine(NugetConfigFile.RepositoryPath, $"{id}.{version}");
            if (!Directory.Exists(dir))
            {
                return;
            }
```

Comment (robin-moss) [reply Assets/NuGet/Editor/NugetHelper.cs:1460]: Ah but you do need relative paths :thinking:, could take the same approach as used when adding support for [native files](https://github.com/GlitchEnzo/NuGetForUnity/pull/421/files#diff-460028a9a166ac23a7e76cf1282330ca51138fc1d7557336de43c60183de8f44R1393)

Comment (robin-moss) [reply Assets/NuGet/Editor/NugetHelper.cs:1460]: To get it working locally I made the following changes

```c#
var dir = Path.Combine(NugetConfigFile.RepositoryPath, $"{id}.{version}");
if (!Directory.Exists(dir))
{
    return;
}
foreach (var analyzer in Directory.GetFiles(dir, "*.dll", SearchOption.AllDirectories)
    .Where(x => x.Contains("/analyzers/")))
{
    // Trigger Unity to create the meta file, the path for ImportAsset MUST be relative to the project
    // path
    var projectRelativePath =
        analyzer.Substring(new DirectoryInfo(Application.dataPath).Parent.FullName.Length + 1);
    AssetDatabase.ImportAsset(projectRelativePath, ImportAssetOptions.ForceSynchronousImport);
    var meta = AssetImporter.GetAtPath(projectRelativePath) as PluginImporter;
```

Comment (nowsprinting) [reply Assets/NuGet/Editor/NugetHelper.cs]: Fixed 9d60da5

**Comment by nowsprinting** [comment]
@robin-moss Thank you for your suggestions.

**Comment by JoC0de** [comment]
I was able to improve the implementation using a combination of your (@nowsprinting) code and things @robin-moss done in #14 
So I am happy with the solution and how it works so I will merge it.
so @nowsprinting Thank you for the PR.
and @robin-moss Thank you for the "improvements".

**Pull request merged**